dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -7,3263 +7,6479 @@
+  }
+  CORBA::ULong length = 1;
+  const DDS::BoundSeq& bound = descriptor->bound();
+  for (CORBA::ULong i = 0; i < bound.length(); ++i) {
+    length *= bound[i];
+  }
+
+  const DDS::DynamicType_var elem_type = get_base_type(descriptor->element_type());
+  const TypeKind elem_tk = elem_type->get_kind();
+  DDS::TypeDescriptor_var elem_td;
+  if (elem_type->get_descriptor(elem_td) != DDS::RETCODE_OK) {
+    return false;
+  }
+
+  if (data_->is_basic_type(elem_tk)) {
+    serialized_size_primitive_array(encoding, size, elem_tk, length);
+    return true;
+  } else if (elem_tk == TK_STRING8) {
+    IndexToIdMap index_to_id(length, MEMBER_ID_INVALID);
+    if (!get_index_to_id_map(index_to_id, length)) {
+      return false;
+    }
+    return serialized_size_generic_string_array<const char*>(encoding, size, index_to_id);
+  } else if (elem_tk == TK_STRING16) {
+#ifdef DDS_HAS_WCHAR
+    IndexToIdMap index_to_id(length, MEMBER_ID_INVALID);
+    if (!get_index_to_id_map(index_to_id, length)) {
+      return false;
+    }
+    return serialized_size_generic_string_array<const CORBA::WChar*>(encoding, size, index_to_id);
+#else
+    return false;
+#endif
+  } else if (elem_tk == TK_ENUM) {
+    const CORBA::ULong bit_bound = elem_td->bound()[0];
+    serialized_size_enum_array(encoding, size, length, bit_bound);
+    return true;
+  } else if (elem_tk == TK_BITMASK) {
+    const CORBA::ULong bit_bound = elem_td->bound()[0];
+    serialized_size_bitmask_array(encoding, size, length, bit_bound);
+    return true;
+  } else if (elem_tk == TK_SEQUENCE) {
+    const DDS::DynamicType_var nested_elem_type = get_base_type(elem_td->element_type());
+    const TypeKind nested_elem_tk = nested_elem_type->get_kind();
+    IndexToIdMap index_to_id(length, MEMBER_ID_INVALID);
+    if (!get_index_to_id_map(index_to_id, length)) {
+      return false;
+    }
+    if (data_->is_basic_type(nested_elem_tk)) {
+      return serialized_size_nesting_basic_array(encoding, size, nested_elem_tk, index_to_id);
+    } else if (nested_elem_tk == TK_ENUM) {
+      return serialized_size_nesting_enum_array(encoding, size, index_to_id);
+    } else if (nested_elem_tk == TK_BITMASK) {
+      return serialized_size_nesting_bitmask_array(encoding, size, index_to_id);
+    }
+  }
+
+  // Elements stored in complex map
+  IndexToIdMap index_to_id;
+  if (!get_index_to_id_from_complex(index_to_id, length)) {
+    return false;
+  }
+  return serialized_size_complex_array(encoding, size, index_to_id, elem_type);
+}
+
+bool DynamicDataImpl::DataContainer::serialize_array(DCPS::Serializer& ser) const
+{
+  DDS::TypeDescriptor_var descriptor;
+  if (type_->get_descriptor(descriptor) != DDS::RETCODE_OK) {
+    return false;
+  }
+  CORBA::ULong length = 1;
+  const DDS::BoundSeq& bound = descriptor->bound();
+  for (CORBA::ULong i = 0; i < bound.length(); ++i) {
+    length *= bound[i];
+  }
+
+  const DDS::DynamicType_var elem_type = get_base_type(descriptor->element_type());
+
```

Comment:
I'm not 100 percent sure about this, but I don't think it matters what the ID is because we know the discriminator will always be first.

Reply:
I don't know why I couldn't find this comment on the file changes view. Could you let me know which lines is this comment for?

Reply:
I can't track it down, line 4809 in the version that GitHub is showing as relevant for this review (a8afeb7cdf4f9d8d896959c5b179eca7d323cf4b) doesn't look like it deals with the member ID of a union's discriminator.

Reply:
I didn't notice this was wrong. I can't imagine I picked the wrong lines, so I guess this is another instance of GitHub (or maybe a combination of this with Firefox?) bugging out on the diff for this file.

There were a few instances of this so this probably isn't the same as what I originally commented on, but they're all the same idea:

https://github.com/objectcomputing/OpenDDS/blob/7ad6752904118825507a4c0fcfe365235a07643f/dds/DCPS/XTypes/DynamicDataImpl.cpp#L6290-L6291